src/contacts/ViewContact/ViewContactContainer.js

```diff
@@ -43,13 +45,14 @@ class ViewContactContainer extends Component {
   };
 
   onClose = () => {
-    const { orgId, mutator } = this.props;
+    const { history, location: { state }, orgId } = this.props;
 
-    mutator.query.replace({
-      _path: orgId ? `/organizations/view/${orgId}` : '/organizations',
-      layer: orgId ? 'edit' : 'create',
-    });
-  };
+    if (orgId) {
+      return (state ? history.goBack() : history.push(`/organizations/view/${orgId}`));
+    } else {
+      return history.push('/organizations/view?layer=create');
```

Comment:
this whole block of code already became complicated. Imagine you will have another one or two places from where you can go to the contact view screen. IMHO, It should be handled via routes to the contact screen

Reply:
This is one of the solutions. I know it is not the best one. But to handle it by routes I need more time to refactor and avoid usage of layers(create or edit).

Reply:
so you prefer code that might not work instead of implementing a solid solution? )

Reply:
Actually no. I just want to say that I need more time for refactoring

Reply:
And I just want to say that this `goBack` si not a solution to go with goBack. I hope we clear here